test-framework/src/main/java/org/apache/brooklyn/test/framework/TestHttpCall.java

```diff
@@ -32,9 +36,34 @@
     @SetFromFlag(nullable = false)
     ConfigKey<String> TARGET_URL = ConfigKeys.newStringConfigKey("url", "URL to test");
 
+    @SetFromFlag(nullable = false)
+    ConfigKey<HttpMethod> TARGET_METHOD = ConfigKeys.builder(HttpMethod.class)
+            .name("method")
+            .description("Method to request the URL: GET, POST, PUT, DELETE, etc")
+            .defaultValue(HttpMethod.GET)
+            .build();
+
+    @SetFromFlag
```

Comment:
I believe you would still be able to set it outside of the `brooklyn.config:` block in yaml. See https://github.com/apache/brooklyn-server/pull/418 for a new test that demonstrates this.

We should get rid of all the `@SetFromFlag`s, where it's not adding additional info (e.g. to say non-null or such like).